File: NhibernateEntity/UserInfo.cs
using System;
using System.Collections;

namespace NhibernateEntity
{
	#region UserInfo

	/// <summary>
	/// UserInfo object for NHibernate mapped table 'UserInfo'.
	/// </summary>
	public class UserInfo
	{
		#region Member Variables
		
		protected int _id;
		protected string _userName;
		protected int _age;
		protected string _address;

		#endregion

		#region Constructors

		public UserInfo() { }

		public UserInfo( string userName, int age, string address )
		{
			this._userName = userName;
			this._age = age;
			this._address = address;
		}

		#endregion

		#region Public Properties

		public int Id
		{
			get {return _id;}
			set {_id = value;}
		}

		public string UserName
		{
			get { return _userName; }
			set
			{
				if ( value != null && value.Length > 50)
					throw new ArgumentOutOfRangeException("Invalid value for UserName", value, value.ToString());
				_userName = value;
			}
		}

		public int Age
		{
			get { return _age; }
			set { _age = value; }
		}

		public string Address
		{
			get { return _address; }
			set
			{
				if ( value != null && value.Length > 50)
					throw new ArgumentOutOfRangeException("Invalid value for Address", value, value.ToString());
				_address = value;
			}
		}

		

		#endregion
	}
	#endregion
}

File: NhibernateEntity/UserInfo.vb
Imports System.Collections

Namespace NhibernateEntity
#Region "UserInfo"

    ''' <summary>
    ''' UserInfo object for NHibernate mapped table 'UserInfo'.
    ''' </summary>
    Public Class UserInfo
#Region "Member Variables"

        Protected _id As Integer
        Protected _userName As String
        Protected _age As Integer
        Protected _address As String

#End Region

#Region "Constructors"

        Public Sub New()
        End Sub

        Public Sub New(ByVal userName As String, ByVal age As Integer, ByVal address As String)
            Me._userName = userName
            Me._age = age
            Me._address = address
        End Sub

#End Region

#Region "Public Properties"

        Public Overridable Property Id() As Integer
            Get
                Return _id
            End Get
            Set(ByVal value As Integer)
                _id = value
            End Set
        End Property

        Public Overridable Property UserName() As String
            Get
                Return _userName
            End Get
            Set(ByVal value As String)
                If value IsNot Nothing AndAlso value.Length > 50 Then
                    Throw New ArgumentOutOfRangeException("Invalid value for UserName", value, value.ToString())
                End If
                _userName = value
            End Set
        End Property

        Public Overridable Property Age() As Integer
            Get
                Return _age
            End Get
            Set(ByVal value As Integer)
                _age = value
            End Set
        End Property

        Public Overridable Property Address() As String
            Get
                Return _address
            End Get
            Set(ByVal value As String)
                If value IsNot Nothing AndAlso value.Length > 50 Then
                    Throw New ArgumentOutOfRangeException("Invalid value for Address", value, value.ToString())
                End If
                _address = value
            End Set
        End Property



#End Region
    End Class
#End Region
End Namespace